com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -3508,6 +3482,28 @@ internal enum TransformExportType { Local, Global, Reset };
                         }
                     }
 
+                    foreach (var unityGo in revisedExportSet)
+                    {
+                        IExportData iData;
+                        if(!exportData.TryGetValue(unityGo, out iData))
+                        {
+                            continue;
+                        }
+                        var data = iData as AnimationOnlyExportData;
```

Comment:
Either cast with `(AnimationOnlyExportData)iData` to throw right here, or test for null (if the 'as' doesn't work).